interlok-core/src/main/java/com/adaptris/core/http/client/net/MultiPartMessageRequestHeaders.java

```diff
@@ -0,0 +1,98 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.adaptris.core.http.client.net;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Enumeration;
+
+import javax.mail.Header;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.commons.lang3.BooleanUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adaptris.annotation.AdapterComponent;
+import com.adaptris.annotation.ComponentProfile;
+import com.adaptris.annotation.InputFieldDefault;
+import com.adaptris.core.AdaptrisMessage;
+import com.adaptris.core.common.MultiPartMessageStreamInputParameter;
+import com.adaptris.core.http.client.RequestHeaderProvider;
+import com.adaptris.util.text.mime.MimeConstants;
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * Implementation of {@link RequestHeaderProvider} that applies multipart MIME message headers as headers to a {@link HttpURLConnection}. It
+ * is meant to be used in conjunction with {@link MultiPartMessageStreamInputParameter}
+ *
+ * @config http-multipart-message-request-headers
+ *
+ */
+@XStreamAlias("http-multipart-message-request-headers")
+@AdapterComponent
+@ComponentProfile(summary = "Use a multipart message headers and add them as headers to an http connection", tag = "mime,http")
+public class MultiPartMessageRequestHeaders extends RequestHeaders implements RequestHeaderProvider<HttpURLConnection> {
+  protected transient Logger log = LoggerFactory.getLogger(this.getClass());
+
+  /**
+   * If set to true that will stop the multipart mime message Message-ID to be sent as an HTTP header. By default it is set to true.
+   *
+   * @param excludeMessageId
+   * @return excludeMessageId
+   */
+  @Getter
+  @Setter
+  @InputFieldDefault("true")
+  private Boolean excludeMessageId;
+
+  public MultiPartMessageRequestHeaders() {
+  }
+
+  @Override
+  public HttpURLConnection addHeaders(AdaptrisMessage msg, HttpURLConnection target) {
+    try {
+      MimeMessage mimeMessage = new MimeMessage(null, msg.getInputStream());
+      Enumeration<Header> headers = mimeMessage.getAllHeaders();
+
+      while (headers.hasMoreElements()) {
+        Header header = headers.nextElement();
+        String value = unfold(header.getValue());
+
+        if (excludeMessageId() && MimeConstants.HEADER_MESSAGE_ID.equals(header.getName())) {
+          log.trace("Ignoring Request Property [{}: {}]", header.getName(), value);
+        } else {
+          log.trace("Adding Request Property [{}: {}]", header.getName(), value);
+          target.addRequestProperty(header.getName(), value);
+        }
+      }
+
+    } catch (MessagingException | IOException expts) {
+      log.error("Invalid multipart MIME message. No request header will be added.", expts);
```

Comment:
Warning instead, since you're not throwing exception, do we need an extra debug flag gives us the exception, or just log it as part of the warning anyway?

Reply:
Actually, no, should probably be a throw new RuntimeException() because it might break the server's expectations about how the payload should look?

Reply:
I guess throwing an error is the right thing to do, even more so because we will rich it only when there is I/O error and not a bad formatting.